internal/cli/atlas/livemigrations/validation/create.go

```diff
@@ -0,0 +1,121 @@
+// Copyright 2021 MongoDB Inc
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package validation
+
+import (
+	"github.com/mongodb/mongocli/internal/cli"
+	"github.com/mongodb/mongocli/internal/config"
+	"github.com/mongodb/mongocli/internal/flag"
+	"github.com/mongodb/mongocli/internal/store"
+	"github.com/mongodb/mongocli/internal/usage"
+	"github.com/openlyinc/pointy"
+	"github.com/spf13/cobra"
+	"go.mongodb.org/atlas/mongodbatlas"
+)
+
+type CreateOpts struct {
+	cli.GlobalOpts
+	cli.OutputOpts
+	destinationClusterName      string
+	destinationDropEnabled      bool
+	migrationHosts              []string
+	sourceCACertificatePath     string
+	sourceClusterName           string
+	sourceProjectID             string
+	sourceSSL                   bool
+	sourceManagedAuthentication bool
+	sourceUserName              string
+	sourcePassword              string
+	store                       store.LiveMigrationValidationsCreator
+}
+
+func (opts *CreateOpts) initStore() error {
+	var err error
+	opts.store, err = store.New(store.AuthenticatedPreset(config.Default()))
+	return err
+}
+
+var createTemplate = `ID	PROJECT ID	SOURCE PROJECT ID	STATUS
+{{.ID}}	{{.GroupID}}	{{.SourceGroupID}}	{{.Status}}`
+
+func (opts *CreateOpts) Run() error {
+	createRequest := opts.newValidationCreateRequest()
+
+	r, err := opts.store.CreateValidation(opts.ConfigProjectID(), createRequest)
+	if err != nil {
+		return err
+	}
+
+	return opts.Print(r)
+}
+
+func (opts *CreateOpts) newValidationCreateRequest() *mongodbatlas.LiveMigration {
+	return &mongodbatlas.LiveMigration{
+		Source: &mongodbatlas.Source{
+			ClusterName:           opts.sourceClusterName,
+			GroupID:               opts.sourceProjectID,
+			Username:              opts.sourceUserName,
+			Password:              opts.sourcePassword,
+			SSL:                   pointy.Bool(opts.sourceSSL),
+			CACertificatePath:     opts.sourceCACertificatePath,
+			ManagedAuthentication: pointy.Bool(opts.sourceManagedAuthentication),
+		},
+		Destination: &mongodbatlas.Destination{
+			ClusterName: opts.destinationClusterName,
+			GroupID:     opts.ConfigProjectID(),
+		},
+		MigrationHosts: opts.migrationHosts,
+		DropEnabled:    pointy.Bool(opts.destinationDropEnabled),
```

Comment:
[nit] you don't need `pointy` here, pointy is only helpful if you have a plain value, `pointy.Bool(true)` as you can't do `&(true)` for all of this you can simply use `&opts.sourceSSL`

Reply:
Thanks, I got carried away with new fun helpers